### DISCUSSION TITLE
Drop PSR-2 coding standard and raise it to PSR-12

### DISCUSSION THREAD

**Discussion by sebastianneubert** [Codeception/Codeception#6324] [Ideas]
## What

In the contribution document it's recommended to use PSR-2, which is deprecated. 

## How

Switch to PSR-12, which is current PHP standard and used by most open source PHP projects.

- add PHP code style fixer and default rules to the root project dir to easily detect code style violations
- add a composer script to easily run the code style fixer
- add a git hook which checks the code style

If you are okay with that idea, I'll like to support Codeception again and will prepare a pull request. 👋

**Comment by Naktibalda** [comment]
This is a very good idea, please do it.

I think that Codeception 4 should be left as is, unless fixing it to comply with PSR-2 (or even PSR-12) requires a small number of changes.
Codeception 5 can be made compliant to PSR-12.
